metadata.yaml

```diff
@@ -19,14 +19,31 @@ tags:
     - monitoring
     - alerting
     - grafana
-series:
-    - kubernetes
+bases:
+    - name: ubuntu
+      channel: 20.04/stable
+containers:
+  prometheus:
+    resource: prometheus-image
+    mounts:
+      - storage: database
+        location: /var/lib/prometheus
+provides:
+  monitoring:
+    interface: prometheus
 requires:
     grafana-source:
         interface: grafana-datasource
     alertmanager:
         interface: alertmanager
+peers:
+  prometheus:
+    interface: prometheus
 storage:
   database:
     type: filesystem
-    location: /var/lib/prometheus
+resources:
+  prometheus-image:
+    type: oci-image
+    description: upstream docker image for prometheus
+    upstream-source: "prom/prometheus:latest"
```

Comment:
the `upstream-source` property does not exist in the schema.

Reply:
Hmm. I do remember seeing that. Will check again and  remove if it is indeed not supported.

Reply:
Looks like there is (or has) been an `upstream-source` property . See for instance

1. https://discourse.charmhub.io/t/writing-a-kubernetes-v1-charm/3976/3
2. https://discourse.charmhub.io/t/specify-image-or-resources-generally-via-url/2426/2

I do not know if this has been deprecated in metadata v2. If it has been it certainly has not been made explicit on the metadata v2 discourse page linked above.